kubernetes/istio-auth.yaml

```diff
@@ -0,0 +1,343 @@
+    rules:
+    - selector: true
+      aspects:
+      - kind: quotas
+        params:
+          quotas:
+          - descriptor_name: RequestCount
+            max_amount: 5
+            expiration: 1s
+      - kind: metrics
+        adapter: prometheus
+        params:
+          metrics:
+          - descriptor_name: request_count
+            # we want to increment this counter by 1 for each unique (source, target, service, method, response_code) tuple
+            value: "1"
+            labels:
+              source: source.name | source.service | "unknown"
+              target: target.service | "unknown"
+              response_code: response.code | response.http.code | 200
+          - descriptor_name:  request_latency
+            value: response.duration | response.latency | "0ms"
+            labels:
+              source: source.name | source.service | "unknown"
+              target: target.service | "unknown"
+              response_code: response.code | response.http.code | 200
+---
+# Mixer
+apiVersion: v1
+kind: Service
+metadata:
+  name: istio-mixer
+  labels:
+    istio: mixer
+spec:
+  ports:
+  - name: tcp
+    port: 9091
+  - name: prometheus
+    port: 42422
+  selector:
+    istio: mixer
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: istio-mixer
+spec:
+  replicas: 1
+  template:
+    metadata:
+      annotations:
+        alpha.istio.io/sidecar: ignore
+      labels:
+        istio: mixer
+    spec:
+      containers:
+      - name: mixer
+        image: docker.io/istio/mixer:2017-04-19-00.29.49
+        imagePullPolicy: Always
+        ports:
+        - containerPort: 9091
+        - containerPort: 9094
+        - containerPort: 42422
+        args:
+          - --globalConfigFile=/etc/opt/mixer/globalconfig.yml
+          - --serviceConfigFile=/etc/opt/mixer/serviceconfig.yml
+          - --logtostderr
+          - -v
+          - "3"
+        volumeMounts:
+          - mountPath: /etc/opt/mixer
+            name: config
+      volumes:
+        - name: config
+          configMap:
+            name: mixer-config
+---
+
+# Permissions and roles for istio
+# To debug: start the cluster with -vmodule=rbac,3 to enable verbose logging on RBAC DENY
+# Also helps to enable logging on apiserver 'wrap' to see the URLs.
+# Each RBAC deny needs to be mapped into a rule for the role.
+# If using minikube, start with '--extra-config=apiserver.Authorization.Mode=RBAC'
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  namespace: default
+  name: istio-manager
+rules:
+- apiGroups: ["istio.io"]
+  resources: ["istioconfigs", "istioconfigs.istio.io"]
+  verbs: ["*"]
+- apiGroups: ["extensions"]
+  resources: ["thirdpartyresources", "thirdpartyresources.extensions", "ingresses"]
+  verbs: ["*"]
+- apiGroups: [""]
+  resources: ["configmaps", "endpoints", "pods","services"]
+  verbs: ["get","watch","list"]
+---
+# Grant permissions to the Manager/discovery.
```

Comment:
It runs in 1.6+ clusters, no need to have RBAC. 
The story would be: for older clusters you can add it manually ( using the individual files and the
old instructions). We focus on 1.6+ because people who care about auth and mutual-TLS auth 
are likely to run with RBAC and recent clusters, and we can focus testing on the future environment.